src/App.tsx

```diff
@@ -1,16 +1,26 @@
 import './App.scss';
+import { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
+
+export const App: React.FC = () => {
+  const [movies, setMovies] = useState(moviesFromServer);
+
+  const onAdd:(movie:Movie)=> void = (movie) => {
+    setMovies(() => (
+      [...movies, movie]
+    ));
```

Comment:
When `setMovies` calls your callBack function, it will pass the `prevState` argument there, this is the reason why we use a callback when we need a `prevState`.
```suggestion
    setMovies((prevState) => (
      [...prevState, movie]
    ));
```